lbrynet/p2p/BlobManager.py

```diff
@@ -39,9 +39,16 @@ def stop(self):
     def get_blob(self, blob_hash, length=None):
         """Return a blob identified by blob_hash, which may be a new blob or a
         blob that is already on the hard disk
+
+        :param str blob_hash: String representing the desired blob's hash
+        :param length: Length of the blob to download. If None is specified, then
+          the blob is assumed to already be saved in the `self.blobs` dict.
+        :raises TypeError: If the blob isn't saved in `self.blobs` and `length` is
+          passed in as a non-`int` type (`None` by default)
```

Comment:
I can add type hinting in the parameter list, though the comments should be kept as a reference for anyone trying to make sense of it in the future.

Reply:
Just keep it as succinct as possible. I prefer to make the method names, the arguments names and the code cleaner rather than to have to provide comments. Comments should be a last resort when the code is too complicated and can't be made simpler.

I've seen too many times where comments were written with good intentions, then someone changed the code and now the comments don't represent the code and so readers end up having to waste time reading incorrect comments and also studying the code anyways to understand what's going on. I believe it's more efficient and a greater win-win to just write clean and simple to understand code rather to have shit code that needs to be commented.

With that caveat, I'm fine if you want to leave some comment, just please keep it short and sweet and to the point and don't include anything obvious that can't be inferred from the method name, arguments names or a quick skim of the code.

Reply:
I agree completely. However, when the function body contains a lot of asynchronous object oriented method calls that aren't formatted in a neat and easily understandable way, anyone trying to read it and maintain it will have to spend way too much time trying to figure out what the hell is going on. 

As a matter of fact, a lot of the variable and method names *are* very well written, and I appreciate that ***very*** much. The problem comes in the fact that when your functions begin exceeding 40 lines of code without any comments, it becomes difficult to understand what's going on. The Google standard states that you should *try* to limit functions to 40 lines, and if you're going to write a long block of code, you should put a block comment explaining what's going on. In most cases a lot of these functions exceed well beyond the 80 lines mark without *any* explanation as to what is going on. 

### Where Comments are Redundant:
```python
def max(a, b):
    # return the larger value
    return a if a > b else b
```

### Where Comments are Useful:
```python
def FFT_vectorized(x):
    """A vectorized, non-recursive version of the Cooley-Tukey FFT"""
    x = np.asarray(x, dtype=float)
    N = x.shape[0]

    if np.log2(N) % 1 > 0:
        raise ValueError("size of x must be a power of 2")

    # N_min here is equivalent to the stopping condition above,
    # and should be a power of 2
    N_min = min(N, 32)
    
    # Perform an O[N^2] DFT on all length-N_min sub-problems at once
    n = np.arange(N_min)
    k = n[:, None]
    M = np.exp(-2j * np.pi * n * k / N_min)
    X = np.dot(M, x.reshape((N_min, -1)))

    # build-up each level of the recursive calculation all at once
    while X.shape[0] < N:
        X_even = X[:, :X.shape[1] / 2]
        X_odd = X[:, X.shape[1] / 2:]
        factor = np.exp(-1j * np.pi * np.arange(X.shape[0])
                        / X.shape[0])[:, None]
        X = np.vstack([X_even + factor * X_odd,
                       X_even - factor * X_odd])
```

Reply:
I think I addressed this pretty clearly in my original comment, again, to reiterate:

> I believe it's more efficient and a greater win-win to just write clean and simple to understand code rather to have shit code that needs to be commented.

You seem to agree but miss the obvious: your criticism of the code is correct, so fix the code, don't compound the technical debt by also adding comments. :-D

I agree that methods longer than 40 lines are too long. Can you fix that? Instead of adding even more lines via comments. <-- the only exception to that is if there is a performance benefit of inlining complicated algorithms in one function, but this is very rarely necessary and in this narrow case a comment is acceptable (to say why the method is massive).

No code is sacred here, if you think something can be done better please do it better. I'm a big fan of less code and simpler code. I have a special place in my heart for PRs that delete code leaving things simpler and cleaner while still functioning as before.

Have you seen this? https://github.com/lbryio/lbry/wiki/Zen-of-LBRY-Development

Reply:
No, I haven't actually. I asked about a standard in #dev, and was told that there is no set standard, and so I just defaulted to the Google standard. 

I wholeheartedly agree with you, though. I think that if you can reduce something and make it simpler and easier to understand, you should. Sometimes the best solutions are the simplest ones, while other times, they are more complicated and thus require a more thorough explanation. This is *why* I want to be adding comments in the first place. This code is extremely messy, and so it helps people to get a grasp and hopefully simplify the process in the first place.

Reply:
At a minimum we use `pylint` and this is actually what failed your PR. You have an unused import (which will get resolved if you use it in a python 3 style annotation).

The standards currently applied / will be applied soon:
1. `pylint` runs automatically on all PRs
2. `mypy` not yet integrated but will be soon (which means use Python 3 type annotations)
3. https://github.com/lbryio/lbry/wiki/Zen-of-LBRY-Development

Reply:
Oh wow, that was stupid of me. I imported that since it references the class that is being passed. I'm just going to use MyPy type hinting since it actually makes use of the import that I did JUST for that comment. 

Thank you for bringing that to my attention, although this PR was very stupid, you've taught a lot in terms of what to do and what not to do. I'm not sure how to edit the PR to put in the correct changes, so I guess I'll just close this one and reopen another one that does it correctly. 

Also you were right, starting out with a small pull request was a good choice once I actually started looking under the hood and playing around with it, so thank you for the advice.

Reply:
Sounds good. Like I said, no code is sacred in lbry, feel free to improve anything you want. We believe in breaking things and moving forward quickly. Looking forward to your future PRs.

Reply:
reply me on slack